R/write_ts.R

```diff
@@ -99,19 +99,21 @@ write_ts <- function(tl,
         freq <- frequency(tl[[1]])
         
         tsdf <- as.data.table(tsmat)
-        tsdf[, date := formatNumericDate(dates, freq, date_format)]
+        tsdf[, t := dates]
+        dates_formatted <- formatNumericDate(dates, freq, date_format)
         
         # Then cbinding xts, index is added as a column. We don't want that.
         # Alternatively: suppressWarnings?
         if("index" %in% names(tsdf)) {
           tsdf <- tsdf[, -"index", with = FALSE]
         }
         
-        setcolorder(tsdf, c(nTs+1, seq(nTs)))
-        setnames(tsdf, c("date", names(tl)))
-        
         if(transpose) {
-          tsdf <- dcast(melt(tsdf, id.vars = "date", variable.name = "series"), series ~ date)
+          tsdf <- dcast(melt(tsdf, id.vars = "t", variable.name = "series"), series ~ t)
```

Comment:
ist this data.table's dcast being used here?

Reply:
We only import `data.table` so I would say yes. Looks like `reshape2`s is only used in the tests.